lib/src/view/relation/following_screen.dart

```diff
@@ -0,0 +1,372 @@
+import 'package:lichess_mobile/src/widgets/feedback.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+
+import 'package:lichess_mobile/src/widgets/platform.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/styles/lichess_colors.dart';
+import 'package:lichess_mobile/src/utils/navigation.dart';
+import 'package:lichess_mobile/src/view/user/user_screen.dart';
+import 'package:lichess_mobile/src/model/user/user.dart';
+
+final RouteObserver<PageRoute<void>> followingRouteObserver =
+    RouteObserver<PageRoute<void>>();
+
+final isInvalidFollowingCacheProvider = StateProvider<bool>((ref) => false);
+final isInvalidFollowingOnlines = StateProvider<bool>((ref) => false);
+
+class FollowingScreen extends ConsumerStatefulWidget {
+  const FollowingScreen({super.key});
+  @override
+  ConsumerState<FollowingScreen> createState() => _FollowingScreenState();
+}
+
+class _FollowingScreenState extends ConsumerState<FollowingScreen>
+    with RouteAware {
+  @override
+  Widget build(BuildContext context) {
+    return PlatformWidget(androidBuilder: _buildAndroid, iosBuilder: _buildIos);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final route = ModalRoute.of(context);
+    if (route != null && route is PageRoute) {
+      followingRouteObserver.subscribe(this, route);
+    }
+  }
+
+  @override
+  void dispose() {
+    followingRouteObserver.unsubscribe(this);
+    super.dispose();
+  }
+
+  @override
+  void didPop() {
+    if (ref.read(isInvalidFollowingOnlines.notifier).state) {
+      ref
+          .read(
+            relationCtrlProvider.notifier,
+          )
+          .getFollowingOnlines();
+      ref.read(isInvalidFollowingOnlines.notifier).state = false;
+    }
+    _invalidateFollowingCache();
+    super.didPop();
+  }
+
+  @override
+  void didPushNext() {
+    _invalidateFollowingCache();
+    super.didPushNext();
+  }
+
+  Widget _buildIos(BuildContext context) {
+    return const CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(),
+      child: _Body(),
+    );
+  }
+
+  Widget _buildAndroid(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.l10n.following),
+      ),
+      body: const _Body(),
+    );
+  }
+
+  void _invalidateFollowingCache() {
+    if (ref.read(isInvalidFollowingCacheProvider.notifier).state) {
+      Future.delayed(const Duration(milliseconds: 500), () {
+        ref.invalidate(followingProvider);
+      });
+      ref.read(isInvalidFollowingCacheProvider.notifier).state = false;
+    }
+  }
+}
+
+class _Body extends ConsumerWidget {
+  const _Body();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final following = ref.watch(followingProvider);
+    final relationState = ref.watch(relationCtrlProvider);
+    final followingOnlines = relationState.when(
```

Comment:
To avoid having 2 `.when()`, which creates branching complexity, it is usually better to combine the async data upstream in a single provider.
As done here for instance: https://github.com/lichess-org/mobile/blob/38fa6bc675f0c580eeced938ff7166e07a101ed1/lib/src/view/puzzle/puzzle_themes_screen.dart#L26